### DISCUSSION TITLE
Issues on array construction

### DISCUSSION THREAD

**Discussion by kumori123** [kind2-mc/kind2#1002] [Q&A] [answered]
Hi, I tried to construct a high-dimension array using code below
```
function Operator1 ( Input1:int8^3^3; Input2:int8 ) returns ( Output1:int8^3^3 )
        var
                _L1:int8^3^3;
                _L2:int8^3^3;
                _L3:int8;

        let
                _L1 = [_L2[0], _L2[1], _L2[2]];
                _L2 = Input1;
                Output1 = _L1;
                _L3 = Input2;
                --%MAIN ;
                check Output1[1][2] > (int8 1);
        tel
```
And I found that KIND2 (I tried 1.5.1 and 2.0)would give errors.
Is this a bug or I got wrong with something?
Thank you!

**Comment by daniel-larraz** [comment, marked as answer]
I recently [fixed](https://github.com/kind2-mc/kind2/pull/999) some bugs related to arrays including the one reported here. The fixes are only available in the develop version. You can either download a [nightly version](https://github.com/kind2-mc/kind2/releases/tag/nightly) of Kind 2, pull the [Docker dev](https://github.com/kind2-mc/kind2#retrieving--updating-the-image) image of Kind 2, or use opam to [pin](https://github.com/kind2-mc/kind2#using-opam) the Kind 2 repository and build Kind 2 with the latest changes.

**Reply by kumori123** [reply]
Thank you!
